### DISCUSSION TITLE
Docker Hub Authentication Issue

### DISCUSSION THREAD

**Discussion by Atomicbeast101** [xelalexv/dregsy#108] [General]
I am struggling to get the authentication to work with Docker Hub registry, Had the `{"username": "...", "password", "..."}` in base64 format as the auth for this registry and it still keeps throwing this error:

> ERRO[0000] error expanding tags: failed listing tags during tag set expansion: error listing image tags for ref 'registry.hub.docker.com/python': time="2023-10-07T12:43:10Z" level=fatal msg="Error listing repository tags: fetching tags list: authentication required"

It's possible that auth isn't needed as I'm pulling a public image from Docker Hub but setting `auth: none` doesn't seem to change the results. 

Does anyone have an active sync in place between Docker Hub as source and any registry target? Mind sharing how you had yours configured to see if I missed anything with mine?

**Comment by xelalexv** [comment]
I've just run the test set, and all tests syncing from/to Docker Hub worked. So at least there don't seem to be any recent changes at Docker Hub that would break the latest dregsy release. Could you run your task with debug log level?

**Reply by Atomicbeast101** [reply]
I was able to figure out the issue. It wasn't authentication related, it was the path to the image that caused it. Before I had this:
```    mappings:
      - from: python
        platform: all
        tags:
          - 'semver: >=3.0.0'
          - 'keep: .+-(slim)'
```

Switching `python` to `library/python` fixed the issue I was having. Didn't realize I have to add the "library" part for any images that didn't have a slash / in it.
